base/complex.jl

```diff
@@ -107,7 +107,7 @@ abs(z::Complex)  = hypot(real(z), imag(z))
 abs2(z::Complex) = real(z)*real(z) + imag(z)*imag(z)
 inv(z::Complex)  = conj(z)/abs2(z)
 inv{T<:Integer}(z::Complex{T}) = inv(float(z))
-sign(z::Complex) = z/abs(z)
+sign(z::Complex) = z==zero(z)?zero(z):z/abs(z)
```

Comment:
`z == 0` shouldn't be an issue since the type conversion should be automatic.

This is still not type stable for `Complex{T<:Integer}` since the division returns a floating point complex number